test/components/tools/test_tool_invoker.py

```diff
@@ -191,6 +191,37 @@ def streaming_callback(chunk: StreamingChunk) -> None:
         assert tool_call_result.origin == tool_call
         assert not tool_call_result.error
 
+    @pytest.mark.asyncio
+    async def test_run_async_with_streaming_callback(self, invoker):
```

Comment:
Could we update these tests to use a ToolInvoker that uses a ThreadPoolExecutor with at least 2 workers? It would be good to test the parallelism aspect. 

So in addition to using at least 2 workers let's also provide 2-3 tool calls for the ToolInvoker to process to also test this parallelism aspect. 

Could we do this for all tests you added?